File: sdks/typescript/src/task_terminal_values.ts
/** Task terminal status values projected without owning task execution. */
export const TaskTerminalStatus = Object.freeze({
  Completed: "completed",
  Failed: "failed",
  Cancelled: "cancelled",
  TimedOut: "timed_out",
  Skipped: "skipped",
} as const);

export type TaskTerminalStatus =
  (typeof TaskTerminalStatus)[keyof typeof TaskTerminalStatus];

export function taskTerminalStatusAsStr(status: TaskTerminalStatus): string {
  if (!Object.values(TaskTerminalStatus).includes(status)) {
    throw new TypeError("invalid task terminal status");
  }
  return status;
}

File: sdks/python/src/echo_agent_sdk/task_terminal_values.py
from __future__ import annotations

from enum import Enum


class TaskTerminalStatus(str, Enum):
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"
    TIMED_OUT = "timed_out"
    SKIPPED = "skipped"

    def as_str(self) -> str:
        return self.value
